src/zenml/integrations/azure/flavors/azureml_orchestrator_flavor.py

```diff
@@ -0,0 +1,160 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the AzureML Orchestrator flavor."""
+
+from typing import TYPE_CHECKING, Optional, Type
+
+from pydantic import Field
+
+from zenml.config.base_settings import BaseSettings
+from zenml.integrations.azure import (
+    AZURE_RESOURCE_TYPE,
+    AZUREML_ORCHESTRATOR_FLAVOR,
+)
+from zenml.models import ServiceConnectorRequirements
+from zenml.orchestrators.base_orchestrator import (
+    BaseOrchestratorConfig,
+    BaseOrchestratorFlavor,
+)
+from zenml.utils.secret_utils import SecretField
+
+if TYPE_CHECKING:
+    from zenml.integrations.azure.orchestrators import AzureMLOrchestrator
+
+
+class AzureMLOrchestratorSettings(BaseSettings):
+    """Settings for the AzureML orchestrator."""
+
+    compute_target: str = Field(
+        description="The name of the compute target to either use."
+    )
+
+
+class AzureMLOrchestratorConfig(
+    BaseOrchestratorConfig, AzureMLOrchestratorSettings
+):
+    """Configuration for the AzureML orchestrator."""
+
+    subscription_id: str = Field(
+        description="Subscription ID that AzureML is running on."
+    )
+    resource_group: str = Field(
+        description="Name of the resource group that AzureML is running on.",
+    )
+    workspace: str = Field(
+        description="Name of the workspace that AzureML is running on."
+    )
+
+    # Service principal authentication
+    # https://docs.microsoft.com/en-us/azure/machine-learning/how-to-setup-authentication#configure-a-service-principal
+    tenant_id: Optional[str] = SecretField(default=None)
```

Comment:
I modeled the implementation after the Sagemaker Orchestrator. If you check its implementation, you will see that it also features configurations both with and without a service connector. For people who are not yet familiar with the concept of service connectors, this might be a bit easier perhaps.

Reply:
I think sagemaker has these because of legacy reasons tbh. It seems to me like the required attributes here are the same as would be required for a service connector, only that the service connector gives additional benefits (e.g. running from the server) and also allows them to more easily connect other components.

I'd personally keep it to just the service connector option (and maybe implicit auth) to have less confusion around these optional attributes. @htahir1 @stefannica any opinion?

Reply:
Yes service connector only please

Reply:
this is removed now.